plugins/discovery-ec2/src/main/java/org/opensearch/discovery/ec2/Ec2DiscoveryPlugin.java

```diff
@@ -161,6 +152,9 @@ static Settings getAvailabilityZoneNodeAttributes(Settings settings, String azMe
         final URLConnection urlConnection;
         try {
             url = new URL(azMetadataUrl);
+            // Obtain the current EC2 instance availability zone from IMDS.
+            // Same as curl http://169.254.169.254/latest/meta-data/placement/availability-zone/.
+            // TODO: use EC2MetadataUtils::getAvailabilityZone that was added in AWS SDK v2 instead of rolling our own
```

Comment:
I think this change effectively will move us from IMDSv1 to IMDSv2. Something we should consider as a fast follow-up